### PR TITLE
Cannot pull from istag referencing image tests

### PR DESCRIPTION
**Pull request by legionus**
Tests for dockerregistry #8326

### CODE REVIEW

**Comment by legionus** [comment]
[test]

**Comment by soltysh** [comment]
LGTM, @legionus can you update this branch and we can get it in?

File: test/end-to-end/core.sh [hunk 1]
```diff
@@ -135,6 +135,48 @@ os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/cache/mysql:pullthrough"
...
+os::cmd::expect_success "oc tag ruby-22-centos7@${tagid} cross:repo-pull-id"
+os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/cache/cross:repo-pull-id"
+echo "[INFO] Cross repository pull successful"
+
+os::cmd::expect_success "docker rmi -f ${imageid}"
+echo "[INFO] Tagging ruby-22-centos7@${tagid} to cross namespace and pulling it"
+os::cmd::expect_success "oc tag cache/ruby-22-centos7@${tagid} cross:namespace-pull-id -n custom"
+os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/custom/cross:namespace-pull-id"
+echo "[INFO] Cross namespace pull successful"
+
```
Comment (miminar) [inline test/end-to-end/core.sh]: Can we also add a test for pulling from chained cross repository? Also we need a test case for referencing image by dockerimage reference as well. Both can be done in a single test:

```
oc import-image --from="${DOCKER_REGISTRY}/custom/cross:namespace-pull" cross-cross:latest
docker pull ${DOCKER_REGISTRY}/cache/cross-cross:latest
```

Comment (soltysh) [inline test/end-to-end/core.sh]: That's a wacky scenario, but agreed it's valuable.

**Comment by soltysh** [comment]
Superseded by https://github.com/openshift/origin/pull/9731

**Pull request closed without merging**